vm/actor/tests/paych_actor_test.rs

```diff
@@ -0,0 +1,1192 @@
+        let lane_param = LaneParams {
+            epoch_num: curr_epoch,
+            from: payer_addr,
+            to: payee_addr,
+            amt: (i + 1) as i64,
+            lane: i as u64,
+            nonce: i + 1,
+        };
+
+        last_sv = require_add_new_lane(&mut rt, lane_param);
+    }
+
+    (rt, last_sv)
+}
+
+fn require_add_new_lane<BS: BlockStore>(
+    rt: &mut MockRuntime<'_, BS>,
+    param: LaneParams,
+) -> SignedVoucher {
+    let payee_addr = Address::new_id(103);
+    let sig = Signature::new_bls(vec![0, 1, 2, 3, 4, 5, 6, 7]);
+    let sv = SignedVoucher {
+        time_lock_min: param.epoch_num,
+        time_lock_max: u64::MAX,
+        secret_pre_image: vec![],
+        extra: Option::None,
+        lane: param.lane,
+        nonce: param.nonce,
+        amount: BigInt::from(param.amt),
+        min_settle_height: 0,
+        merges: vec![],
+        signature: Some(sig.clone()),
+    };
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), param.from);
+    rt.expect_validate_caller_addr(&[param.from, param.to]);
+    rt.expect_verify_signature(sig.clone(), payee_addr, to_vec(&sv).unwrap(), ExitCode::Ok);
+
+    let ucp = UpdateChannelStateParams {
+        proof: vec![],
+        secret: vec![],
+        sv: sv,
+    };
+
+    assert!(rt
+        .call(
+            &PAYCH_ACTOR_CODE_ID.clone(),
+            Method::UpdateChannelState as u64,
+            &Serialized::serialize(ucp).unwrap(),
+        )
+        .is_ok());
+
+    rt.verify();
+    SignedVoucher {
+        time_lock_min: param.epoch_num,
+        time_lock_max: u64::MAX,
+        secret_pre_image: vec![],
+        extra: Option::None,
+        lane: param.lane,
+        nonce: param.nonce,
+        amount: BigInt::from(param.amt),
+        min_settle_height: 0,
+        merges: vec![],
+        signature: Some(sig.clone()),
+    }
+}
+
+fn construct_and_verify<BS: BlockStore>(
+    rt: &mut MockRuntime<'_, BS>,
+    sender: Address,
+    receiver: Address,
+) {
+    let params = ConstructorParams {
+        from: sender,
+        to: receiver,
+    };
+    rt.expect_validate_caller_type(&[INIT_ACTOR_CODE_ID.clone()]);
+    assert!(rt
+        .call(
+            &PAYCH_ACTOR_CODE_ID,
+            METHOD_CONSTRUCTOR,
+            &Serialized::serialize(&params).unwrap(),
+        )
+        .is_ok());
+    rt.verify();
+    verify_initial_state(rt, sender, receiver);
+}
+
+fn verify_initial_state<BS: BlockStore>(
+    rt: &mut MockRuntime<'_, BS>,
+    sender: Address,
+    receiver: Address,
+) {
+    let state: PState = rt.get_state().unwrap();
+    let expected_state = PState::new(sender, receiver);
+    assert_eq!(expected_state.to, state.to);
+    assert_eq!(expected_state.from, state.from);
+    assert_eq!(expected_state.min_settle_height, state.min_settle_height);
+    assert_eq!(expected_state.settling_at, state.settling_at);
+    assert_eq!(expected_state.to_send, state.to_send);
+    assert_eq!(state.lane_states.len(), 0);
```

Comment:
You're not calling verify_state here, and it seems like you should? (Seems like some of your tests may be wrong, as  `TestActor_UpdateChannelStateRedeem/redeeming_voucher_updates_correctly_with_one_lane` and `TestActor_UpdateChannelStateMergeSuccess` equivalents should have failed with this setup)

Seems merge success test doesn't even call verify_state and it looks like it should?

Reply:
Yeah I just manually inlined the function since it was a fixed call with -1. Shouldve realised it made it harder to compare with Lotus